### ISSUE TITLE
Assigning a dict literal to a list variable in a function raises no issue

### ISSUE THREAD

**Issue by chapel582** [python/mypy#12171]
**Bug Report**

(A clear and concise description of what the bug is.)

**To Reproduce**

The following code reproduces the issue
`from typing import List

class Foo():
	def __init__(self):
		pass

def test_make_update_sql_multiple():
    where_args: List[Foo] = {"column1": ("column1", "oldvalue")}
    print(where_args)`

**Expected Behavior**

The following code produces the expected behavior (raise an error with assigning a dict to a List)

`
from typing import List

class Foo():
	def __init__(self):
		pass

def test_make_update_sql_multiple():
    where_args: List[Foo] = {"any": "thing"}
    print(where_args)
`

(Write what you thought would happen.)

**Actual Behavior**

Mypy sees no problem

(Write what happened.)

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.9.5
- Operating system and version: Windows 10.0.19044 Build 19044

**Comment by erictraut** [comment]
Has there been any thought of enabling "--check-untyped-defs" by default? Someone seems to trip on this at least once a week.

**Comment by JelleZijlstra** [comment]
Yes, I agree the current behavior is not at all intuitive. #11201 lists it as a possible change.
